src/world/generator/PopulationTask.php

```
@@ -29,130 +9,91 @@
+    public function onRun(): void {
+        $context = ThreadLocalGeneratorContext::fetch($this->worldId)
+            ?? throw new AssumptionFailedError("Generator context not initialized");
+
+        $gen = $context->getGenerator();
+        $mgr = new SimpleChunkManager($context->getWorldMinY(), $context->getWorldMaxY());
+
+        $center = $this->centerChunk !== null ? FastChunkSerializer::deserializeTerrain($this->centerChunk) : null;
+        $adjacent = igbinary_unserialize($this->adjacentChunksData);
+        $adjChunks = [];
+
+        foreach ($adjacent as $hash => $data) {
```

PHPStan (2.1.15)

Argument of an invalid type mixed supplied for foreach, only iterables are supported.


(foreach.nonIterable)